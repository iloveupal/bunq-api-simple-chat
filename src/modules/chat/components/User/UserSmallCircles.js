import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import styled from 'styled-components';

import UserColorCircle from 'Modules/chat/components/User/UserColorCircle';

import { getUsers } from 'Modules/chat/domains/users/UsersSelectors';


const CirclesContainer = styled.div`
    display: flex;
    flex-flow: row nowrap;
    padding-top: 5px;
`;

const ItemContainer = styled.div`
    padding-right: 5px;
`;

const mapStateToProps = (state) => ({
    usersData: getUsers(state),
});


export class UserSmallCircles extends PureComponent {
    static propTypes = {
        users: PropTypes.array.isRequired,
        usersData: PropTypes.object.isRequired,
    }

    render () {
        return (
            <CirclesContainer>
                {
                    this.props.users.map(( { userid } ) => (
                        <ItemContainer key={userid}>
                            <UserColorCircle
                                id={userid}
                                name={this.props.usersData[userid].name}
                                small
                            />
                        </ItemContainer>
                    ))
                }
            </CirclesContainer>
        )
    }
}

export default connect(mapStateToProps)(UserSmallCircles);