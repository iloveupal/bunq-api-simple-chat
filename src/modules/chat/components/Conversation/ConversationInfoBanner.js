import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import styled from 'styled-components';

import {
    getCurrentConversationObject,
} from 'Modules/chat/domains/conversations/ConversationsSelectors';

import {
    renderConversationName,
    renderConversationType,
} from 'Modules/chat/domains/conversations/ConversationsUtils';
import UserSmallCircles from 'Modules/chat/components/User/UserSmallCircles';


const Container = styled.div`
    display: flex;
    flex-flow: row nowrap;
    align-items: center;
`;

const NameAndTypeContainer = styled.div`
    display: flex;
    flex-flow: column nowrap;
    padding-right: 15px;
`;

const Name = styled.div`
    font-family: sans-serif;
    font-size: 13px;
    color: white;
    font-weight: bold;
`;

const Type = styled.div`
    font-family: sans-serif;
    font-size: 13px;
    color: #ccc;
`;



class ConversationInfoBanner extends PureComponent {
    static propTypes = {
        data: PropTypes.object,
    };

    render() {
        if ( !this.props.data ) {
            return null;
        }

        return (
            <Container>
                <NameAndTypeContainer>
                    <Name>{renderConversationName(this.props.data)}</Name>
                    <Type>{renderConversationType(this.props.data)}</Type>
                </NameAndTypeContainer>
                <UserSmallCircles users={this.props.data.users} />
            </Container>
        );
    }
}

const mapStateToProps = (state) => ({
    data: getCurrentConversationObject(state),
});

export default connect(mapStateToProps)(ConversationInfoBanner);