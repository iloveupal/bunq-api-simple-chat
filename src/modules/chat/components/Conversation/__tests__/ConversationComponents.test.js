import React from 'react';
import { shallow } from 'enzyme';

import { ConversationInfoBanner, Name, Type } from '../ConversationInfoBanner';
import UserSmallCircles from 'Modules/chat/components/User/UserSmallCircles';



describe('ConversationInfoBanner', () => {
    it('should match snapshots', () => {
        const component = shallow(
            <ConversationInfoBanner
                data={{
                    conversation: {
                        type: '2',
                        name: 'Test',
                    },
                    users: [{ userid: '1' }, { userid: '2' }],
                }}
            />
        );

        expect(component).toMatchSnapshot();
    });

    it('should render nothing when no data is passed', () => {
        const component = shallow(
            <ConversationInfoBanner
                data={null}
            />
        );

        expect(component.children()).toHaveLength(0);

    });

    it('should render correctly when data is passed', () => {
        const component = shallow(
            <ConversationInfoBanner
                data={{
                    conversation: {
                        type: '2',
                        name: 'Test',
                    },
                    users: [{ userid: '1' }, { userid: '2' }],
                }}
            />
        );

        expect(component.find(Name).at(0).text()).toEqual('Test');
        expect(component.find(Type).at(0).text()).toEqual('Group');
        expect(component.find(UserSmallCircles).at(0).props().users).toEqual([{ userid: '1' }, { userid: '2' }]);
    });
});