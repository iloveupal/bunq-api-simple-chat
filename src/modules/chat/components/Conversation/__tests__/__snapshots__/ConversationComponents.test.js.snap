// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationInfoBanner should match snapshots 1`] = `
<ConversationInfoBanner__Container>
  <ConversationInfoBanner__NameAndTypeContainer>
    <ConversationInfoBanner__Name>
      Test
    </ConversationInfoBanner__Name>
    <ConversationInfoBanner__Type>
      Group
    </ConversationInfoBanner__Type>
  </ConversationInfoBanner__NameAndTypeContainer>
  <Connect(UserSmallCircles)
    users={
      Array [
        Object {
          "userid": "1",
        },
        Object {
          "userid": "2",
        },
      ]
    }
  />
</ConversationInfoBanner__Container>
`;
